src/robotFactory.js

```diff
@@ -1,10 +1,101 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this.coords.y;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this.coords.y;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this.coords.x;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this.coords.x;
+  }
+
+  getInfo() {
+    // These two variables are to avoid eslintmax-len error
+    const n = this.name;
+    const cV = this.chipVersion;
+
+    return `Robot: ${n}, Chip version: ${cV}, Weight: ${this.weight}`;
```

Comment:
```suggestion
    return `Robot: ${n}, `
    + `Chip version: ${cV}, ` 
    + `Weight: ${this.weight}`;
```
This works too i think :)

Reply:
Yes you are right :) I will change it. Thanks